app/controllers/users/settings/user_settings_controller.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Users
+  module Settings
+    class UserSettingsController < ApplicationController
+      def show
+        render json: { data: current_user.settings[params[:key]] }
+      end
+
+      def update
+        key = params[:key]
```

Comment:
possibly we need a whitelist of allowed keys

Reply:
It will be not possible, in future here will be dynamic keys based on object ids